shop/src/main/java/ro/msg/learning/shop/service/OrderService.java

```diff
@@ -0,0 +1,81 @@
+package ro.msg.learning.shop.service;
+
+import jakarta.persistence.EntityNotFoundException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import ro.msg.learning.shop.domain.*;
+import ro.msg.learning.shop.dto.OrderDto;
+import ro.msg.learning.shop.dto.ProductIdAndQuantity;
+import ro.msg.learning.shop.mapper.OrderMapper;
+import ro.msg.learning.shop.repository.OrderRepository;
+
+import java.util.List;
+import java.util.UUID;
+
+@Service
+public class OrderService {
+    public static final String ID_INVALID = "ID invalid.";
+    public static final String NO_SUCH_LOCATION = "No such location";
+    public static final String NO_SUCH_CUSTOMER = "No such customer";
+
+    @Autowired
+    private StockService stockService;
+
+    @Autowired
+    private LocationService locationService;
+
+    @Autowired
+    CustomerService customerService;
+
+    @Autowired
+    private OrderRepository orderRepository;
+
+    @Autowired
+    private OrderMapper orderMapper;
+
+
+    private UUID findLocationWithRequiredProducts(OrderDto order) {
+        List<Location> allLocations = locationService.getAllLocations();
+        for (Location location : allLocations) {
+            boolean hasAllProducts = true;
+            for (ProductIdAndQuantity requiredProduct : order.getProductIdAndQuantityList()) {
+                Stock stock = stockService.findById(StockId.builder().location(location.getId()).product(requiredProduct.getProductId()).build()).orElse(null);
+                if (stock == null || stock.getQuantity() < requiredProduct.getQuantity()) {
+                    hasAllProducts = false;
+                    break;
+                }
+            }
+            if (hasAllProducts) {
+                return location.getId();
+            }
+        }
+        return null;
```

Comment:
returning null is discouraged

Reply:
optimized the method (not returning null now)